docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.15.3.rst

```diff
@@ -0,0 +1,199 @@
+
+Major Analysis Improvements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Swift
+"""""
+
+*   Added Swift 5.9.1 support
+*   New AST node is extracted: :code:`SingleValueStmtExpr`
+
+Minor Analysis Improvements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+C/C++
+"""""
+
+*   The "Returning stack-allocated memory" (:code:`cpp/return-stack-allocated-memory`) query now also detects returning stack-allocated memory allocated by calls to :code:`alloca`, :code:`strdupa`, and :code:`strndupa`.
+*   Added models for :code:`strlcpy` and :code:`strlcat`.
+*   Added models for the :code:`sprintf` variants from the :code:`StrSafe.h` header.
+*   Added SQL API models for :code:`ODBC`.
+*   Added taint models for :code:`realloc` and related functions.
+
+C#
+""
+
+*   The predicate :code:`UnboundGeneric::getName` now prints the number of type parameters as a :code:` `N` suffix, instead of a :code:`<,...,>` suffix. For example, the unbound generic type
+    :code:`System.Collections.Generic.IList<T>` is printed as :code:`IList`1` instead of :code:`IList<>`.
+*   The predicates :code:`hasQualifiedName`, :code:`getQualifiedName`, and :code:`getQualifiedNameWithTypes` have been deprecated, and are instead replaced by :code:`hasFullyQualifiedName`, :code:`getFullyQualifiedName`, and :code:`getFullyQualifiedNameWithTypes`, respectively. The new predicates use the same format for unbound generic types as mentioned above.
+*   These changes also affect models-as-data rows that refer to a field or a property belonging to a generic type. For example, instead of writing
+
+    ..  code-block:: yaml
+    
+        extensions:
+          - addsTo:
+              pack: codeql/csharp-all
+              extensible: summaryModel
+              data:
+                - ["System.Collections.Generic", "Dictionary<TKey,TValue>", False, "Add", "(System.Collections.Generic.KeyValuePair<TKey,TValue>)", "", "Argument[0].Property[System.Collections.Generic.KeyValuePair<,>.Key]", "Argument[this].Element.Property[System.Collections.Generic.KeyValuePair<,>.Key]", "value", "manual"]
+
+*   The models-as-data format for types and methods with type parameters has been changed to include the names of the type parameters. For example, instead of writing
+
+    ..  code-block:: yaml
+    
+        extensions:
+          - addsTo:
+              pack: codeql/csharp-all
+              extensible: summaryModel
+              data:
+                - ["System.Collections.Generic", "IList<>", True, "Insert", "(System.Int32,T)", "", "Argument[1]", "Argument[this].Element", "value", "manual"]
+                - ["System.Linq", "Enumerable", False, "Select<,>", "(System.Collections.Generic.IEnumerable<TSource>,System.Func<TSource,System.Int32,TResult>)", "", "Argument[0].Element", "Argument[1].Parameter[0]", "value", "manual"]
+
+Golang
+""""""
+
+*   Added the `gin cors <https://github.com/gin-contrib/cors>`__ library to the CorsMisconfiguration.ql query
```

Comment:
This doesn't seem to be how we normally refer to queries. Usually we seem to use the ID, sometimes with the `name`.

Reply:
Agreed. Should I change it in `go/ql/lib/change-notes/released/0.7.3.md` or `go/ql/lib/CHANGELOG.md` (or both)? Another question: this is actually an experimental query, so I'm not sure it warrants a change note at all. Is that right?

Reply:
That's a good point. @turbo - what do you think?

Reply:
I'm actually happy for the info to be included, it is a change after all that some users (of the experimental suite) may be affected by. The consistency in how we refer to queries is still good feedback.